### PR TITLE
feat(agw): add ho algorithm parameter

### PR DESCRIPTION
**Pull request by sxkmg**
Signed-off-by: root <369888042@qq.com>

 feat(agw): add ho algorithm parameter
   

## Summary

add None 243 Ho  algorithm parameter

## Test Plan

 ##1.Start the magma service
2.Configure the base station none-243 to connect to the gateway
3.Test and issue parameters through the magma swagger API(https://magma-test.localhost:9443/swagger/v1/ui/lte#/EnodeBs/get_lte__network_id__enodebs__enodeb_serial_)
1631007352343_F143C943-8C85-4825-BC5E-BAD96161522E
4.View the configuration file and verify whether the parameters are distributed to the gateway
（ sudo tail -1000f /var/opt/magma/configs/gateway.mconfig ）
lQLPDhqt8XIw6jjNAlzNBGiwiJoxtZlFWOSixRPGnvC3pQ_1128_604

5.Check whether the parameters are successfully distributed to the base station
sudo tail -1000f /var/log/enodebd.log
lALPD4PvOODaWJHNAZXNBms_1643_405

image

### CODE REVIEW

**Comment by sebathomas** [comment]
I think you can solve at least some of the Python lint issues by executing the formatter `./precommit.py --format --diff` inside `lte/gateway/python`.

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 1]
```diff
@@ -2460,6 +2460,159 @@ definitions:
...
+        format: uint32
+        minimum: 0
+        maximum: 63
+        example: 8
+      qrxlevmin_sib1:
+        type: integer
+        format: int32
+        minimum: -70
+        maximum: -22
+        example: 8
```
Comment (alexzurbonsen) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: The example value doesn't match the allowed range. Probably good to align?

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 2]
```diff
@@ -2460,6 +2460,159 @@ definitions:
...
+        format: uint32
+        minimum: 0
+        maximum: 31
+        example: 8
+      qrxlevmin_sib3:
+        type: integer
+        format: int32
+        minimum: -70
+        maximum: -22
+        example: 8
```
Comment (alexzurbonsen) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: Same as above (example value and allowed range don't match)

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 3]
```diff
@@ -2460,6 +2460,159 @@ definitions:
...
+        format: uint32
+        example: 8
+      ciphering_algorithm:
+        type: string
+        example: "128-EEA1,128-EEA2"
+        default: "128-EEA1,128-EEA2"
+      integrity_algorithm:
+        type: string
+        example: "128-EIA1,128-EIA2"
+        default: "128-EIA1,128-EIA2"
```
Comment (alexzurbonsen) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: I think setting default values on this level probably doesn't work (see this [issue](https://github.com/go-swagger/go-swagger/issues/1393)). Instead you could set them one level higher, as described in the issue.

If it works you will see (in the swagger GUI) a line "Default value : ..."  next to the respective value  of the swagger endpoint.

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 1 above)
Comment (sxkmg) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: Sorry, I'll deal with it as soon as possible. Thank you

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 2 above)
Comment (sxkmg) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: Sorry, I'll deal with it as soon as possible. Thank you

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 1 above)
Comment (sxkmg) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: image
Modified, thank you!

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 2 above)
Comment (sxkmg) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: image
Modified, thank you!

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 3 above)
Comment (sxkmg) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: image
The default configuration parameter has been canceled. I don't think it is necessary to configure the default value in this place

**Comment by sxkmg** [comment]
> I think you can solve at least some of the Python lint issues by executing the formatter `./precommit.py --format --diff` inside `lte/gateway/python`.

Ok,thank you!

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 3 above)
Comment (alexzurbonsen) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: Alright!

**Comment by maxhbr** [comment]
Hey @sxkmg , your commits need to have a signoff

**Pull request closed without merging**